frontend/src/features/verifiable-fields/VerifiableFieldProvider.tsx

```diff
@@ -72,65 +94,72 @@ export const VerifiableFieldProvider = ({
         { message: 'Please fill in field before attempting verification' },
         { shouldFocus: true },
       )
-    } else {
-      clearErrors(schema._id)
     }
 
-    // Do nothing if box is already opened.
-    if (isVfnBoxOpen) return
+    // Do nothing if box is already opened, or there is already a signature linked to the input.
+    if (isVfnBoxOpen || mapNumberToSignature[currentInputValue]) return
 
     // Check is valid phone number
     if (isMobilePhoneNumber(currentInputValue)) {
-      // TODO: Call mutation to send SMS.
-      setIsVfnBoxOpen(true)
-    } else {
-      return setError(
-        schema._id,
-        { message: 'Please enter a valid mobile number' },
-        { shouldFocus: true },
-      )
+      return triggerSendOtpMutation.mutate(currentInputValue, {
+        onSuccess: () => setIsVfnBoxOpen(true),
+      })
     }
-  }, [clearErrors, getValues, isVfnBoxOpen, schema._id, setError])
-
-  const handleVfnSuccess = useCallback(
-    async (signature: string) => {
-      const currentValue = getValues(schema._id)?.value
-      if (!currentValue) return
 
-      setValue(
-        schema._id,
-        {
-          value: currentValue,
-          signature,
+    // Else invalid input.
+    return setError(
+      schema._id,
+      { message: 'Please enter a valid mobile number' },
+      { shouldFocus: true },
+    )
+  }, [
+    getValues,
+    isVfnBoxOpen,
+    mapNumberToSignature,
+    schema._id,
+    setError,
+    triggerSendOtpMutation,
+  ])
+
+  const handleVerifyOtp = useCallback(
+    (otp: string) => {
+      // async so verification box can show error message
+      return verifyOtpMutation.mutateAsync(otp, {
+        onSuccess: (signature) => {
+          const currentValue = getValues(schema._id)?.value
+          if (!currentValue) return
+          setValue(
+            schema._id,
+            {
+              value: currentValue,
+              signature,
+            },
+            { shouldValidate: true },
+          )
+          // Add signature to map.
+          setMapNumberToSignature((prev) => ({
```

Comment:
also realised this map will keep signatures beyond 4h, i.e. we will show answers as being verified even though they will not be accepted by the backend